enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -95,263 +112,214 @@ This is an example of an OKD cluster hosting solution where several organization
+* A pipeline associates multiple input names with multiple output names.
+* Users cannot define new inputs (for initial release)
+* Users can define new outputs, output configuration includes:
+  - `type` (e.g. `Syslog`, `Fluent`)
+  - `destination` URL for the endpoint.
+  - `secretRef` referring to a secret object used for secure connections.
+    Secrets not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Forwarding is enabled when a `ClusterLogForwarder` is created, and disabled when it is deleted.
+  - With no `ClusterLogForwarder`: the default Elasticsearch instance is *enabled* (status quo)
+  - With a `ClusterLogForwarder`: the default Elasticsearch instance is *disabled* unless explicitly enabled by a `pipeline` with output`Default`.
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
 
-#### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
+* Server-authenticated TLS is enabled if `destination` is a secure URL (e.g. 'https:')
+* Client-authenticated TLS is enabled if `destination` is secure *and* `secretRef` has keys `tls.crt`, `tls.key`, `ca.crt`
+  - it is an error if `secretRef` is present but `destination` is *not* secure, or the required 'tls.' keys are missing or invalid.
+* An intentionally insecure output (no TLS) must have `insecure: true`
+  - avoid accidental insecure configuration of an output that was intended to be secure.
+* The user is responsible for creating and maintaining the secret objects
+* The cluster logging operator is responsible for watching secrets and applying changes,  e.g. if the user replaces a certificate or changes a password in a secret, the operator must re-connect affected outputs with the new credentials.
+
+#### Reliability
+* Output connections will automatically re-connect on disconnect.
+* Changes to secrets will trigger automatic re-connect with new credentials.
+
+#### Scale
+* The cluster logging operator generates configuration for the collector (fluentd)
+  - with a singleton ClusterLogForwarder this is unlikely to be a scaling problem.
+* The actual connection and forwarding to remote endpoints is done by fluentd
+  - we rely on fluentd to scale and perform.
+
+#### Metrics
+
+Note: the actual forwarding is done by the collector (fluentd) so we can only provide metrics if they are available from fluentd.
+
+Relevant metrics include:
```

Comment:
Does our fluent metrics pipeline provide these? if not it may be a stretch for us to do this